### ISSUE TITLE
🐛 BUG: Can't stop typescript errors

### ISSUE THREAD

**Issue by cdtut** [withastro/language-tools#706]
### Describe the Bug

Can't stop typescript errors appearing.

### Steps to Reproduce

Front matter

```js
const array = [
  { required: '' },
  { required: '', optional: '' }
]
array[1].optional
```

`optional` is underlined in red and hovering explains error `Property 'optional' does not exist on type '{ required: string; optional?: undefined; }'.ts(2339)`

I tried to disable validation with `"checkJs": false` in jsconfig.json and `"typescript.validate.enable": false` in settings but it doesn't stop errors.

No error putting that code in .js file but error still in .astro file.

Using typescript is not requirement for astro so should not be required to write any typescript to stop these errors. They shouldn't appear if typescript checks are disabled.

**Comment by Princesseuh** [comment]
Astro is TypeScript only at this time, TypeScript cannot be disabled.

**Comment by cdtut** [comment]
Instead of `"typescript.validate.enable": false` can an extension specific setting be added to settings file disabling typescript errors in astro files? Can be people who don't want disable all typescript validation if they're writing typescript in non astro files but are writing astro as javascript (without type annotations which is valid typescript).

**Comment by Princesseuh** [comment]
We'd rather make that option work if that's a possibility. Q

**Comment by cdtut** [comment]
Is there any workaround until fixed? Disruptive for developing and causes to miss important errors because so many things are underlined in red. Even every access under Astro.locals complains with error that property doesn't exist.

**Comment by cdtut** [comment]
@Princesseuh Can it be P1 bug? There are so many error everywhere. I can give more example. Cannot trust errors in IDE anymore because most of them are because of this extension instead of real problems with code. Trusting errors is more than nice to have I believe, it's very frustrating developer experience.

**Comment by Princesseuh** [comment]
> @Princesseuh Can it be P1 bug? There are so many error everywhere. I can give more example. Cannot trust errors in IDE anymore because most of them are because of this extension instead of real problems with code. Trusting errors is more than nice to have I believe, it's very frustrating developer experience.

TypeScript errors are real errors.. I'm sorry this is affecting your experience so much, but it's a niche issue, people typically use TypeScript just fine.

**Comment by cdtut** [comment]
> TypeScript errors are real errors..

This is opinion and objectively wrong. There are many example that run and build fine but this extension still complain with error. Rule of thumb is if it runs without error in astro and browser **it's not an error**.

> it's a niche issue, people typically use TypeScript just fine

People typically use javascript just fine. It is allowed to write vanilla javascript in astro and everyone should respect right to choose for own code.

**Comment by Princesseuh** [comment]
Fixed by https://github.com/volarjs/services/commit/6f13b47fc01e9999f6fa46023f80225957f421f8, we'll release an update with it soon-ish

**Comment by cdtut** [comment]
Is it in 2.6.2?

**Comment by Princesseuh** [comment]
Nope, not yet sorry. There was an unexpected delay, unfortunately. The holidays period also delayed things a fair amount. Still definitely planned, though!

**Comment by cdtut** [comment]
Is it in 2.7.1?

**Comment by Princesseuh** [comment]
Yes, it should work now!

**Comment by cdtut** [comment]
I still have typescript errors like `Property 'item' does not exist on type 'Element'. ts(2339)`. Is enough to set "typescript.validate.enable": false`?

**Comment by cdtut** [comment]
@Princesseuh working for you?

**Comment by Princesseuh** [comment]
Ah, it only seems to be working for certain kinds of errors, I'll take a look. Sorry for the inconvenience.

**Comment by cdtut** [comment]
@Princesseuh can you open this issue?

**Comment by Princesseuh** [comment]
I'll open an issue on the upstream project instead, since the code for this isn't in this repo

**Comment by cdtut** [comment]
@Princesseuh where to find that issue number?

**Comment by Princesseuh** [comment]
The option works for me at the moment:
![image](https://github.com/withastro/language-tools/assets/3019731/ab5ef4fe-dedc-49f4-8c31-83c9e3595e69)
